ddsp/training/docker/ddsp_ai_platform.py

```diff
@@ -0,0 +1,270 @@
+    project_id = input('Project ID: ')
+    project_id = check_project_id(project_id)
+  return project_id
+
+def get_region():
+  region = subprocess.getoutput('gcloud config get-value compute/region')
+  if region == '(unset)':
+    region = prompt_value('Region for training the model', 'europe-west4')
+  return region
+
+def get_input():
+  """Gathers input from user"""
+  if FLAGS.data_path:
+    FLAGS.data_path = check_bucket(FLAGS.data_path,
+                                   'Path to training dataset directory: ')
+  else:
+    FLAGS.data_path = prompt_gs_path('Path to training dataset directory',
+                                     required=True)
+
+  if FLAGS.save_dir:
+    FLAGS.save_dir = check_bucket(FLAGS.save_dir,
+                                  'Path for saving model, snapshots '
+                                  'and summaries: ')
+  else:
+    FLAGS.save_dir = prompt_gs_path('Path for saving model, snapshots '
+                                    'and summaries', required=True)
+
+  if not FLAGS.restore_dir:
+    FLAGS.restore_dir = prompt_gs_path('Path for restoring checkpoints '
+                                       'before training')
+    if not FLAGS.restore_dir:
+      FLAGS.restore_dir = FLAGS.save_dir
+
+  if FLAGS.config_path:
+    FLAGS.config_path = check_local_file(FLAGS.config_path,
+                                         'Path to configuration file: ')
+  else:
+    FLAGS.config_path = prompt_local_path('Path to configuration file: ')
+
+  if not FLAGS.project_id:
+    FLAGS.project_id = get_project_id()
+
+  image_uri = 'gcr.io/' + FLAGS.project_id + '/ddsp_training:train_job'
+
+  job_name = 'training_job_' + str(int((datetime.datetime.now()\
+             - datetime.datetime(1970,1,1)).total_seconds()))
+
+  if not FLAGS.region:
+    FLAGS.region = get_region()
+
+  if not FLAGS.batch_size:
+    FLAGS.batch_size = prompt_value('Batch size', '128')
+
+  if not FLAGS.learning_rate:
+    FLAGS.learning_rate = prompt_value('Learning rate', '0.001')
+
+  if not FLAGS.num_steps:
+    FLAGS.num_steps = prompt_value('Number of steps', '15000')
+
+  if not FLAGS.steps_per_save:
+    FLAGS.steps_per_save = prompt_value('Steps per save', '300')
+
+  if not FLAGS.steps_per_summary:
+    FLAGS.steps_per_summary = prompt_value('Steps per summary', '300')
+
+  if not FLAGS.early_stop_loss_value:
+    FLAGS.early_stop_loss_value = prompt_value('Early stop loss value', '5')
+
+  return image_uri, job_name
+
+def build_image(image_uri):
+  """Builds the docker image"""
+  build_command = 'docker build -f Dockerfile -t '\
+                  + image_uri + ' ./'
+  os.system(build_command)
+
+def push_image(image_uri):
+  """Pushes the docker image on Google Cloud Registry"""
+  pushing_image = 'docker push ' + image_uri
+  os.system(pushing_image)
+
+def submit_job(image_uri, job_name):
+  """Submits the job on AI Platform"""
+  os.system('export PATH=/usr/local/google/home/$USER/.local/bin:$PATH')
```

Comment:
Is this true for all systems (mac, linux?)? Is it a path installed by cloud?

Reply:
This is the path for tensorboard command. I added this to the "Before you begin" part of the documentation, instead of leaving it in the script.